herd/monad.mli

```diff
@@ -88,6 +98,12 @@ module type S =
     (* Data composition, entry for snd monad: minimals for complete iico *)
     val bind_data_to_minimals : 'a t -> ('a -> 'b t) -> ('b) t
 
+    val bind_data_to_output : 'a t -> ('a -> 'b t) -> 'b t
+    (** [bind_data_to_output p1 p2] returns a composition of the event
+        structures of [s1] and [s2] where there is iico_causality_data
+        relation from the output events of [s1] to the output events
```

Comment:
I know I wrote this, but I like the comment you added for data_output_union more. Would you like to change this as well?